jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/paths/PathComparator.java

```diff
@@ -0,0 +1,33 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.paths;
+
+import java.nio.file.Path;
+import java.util.Comparator;
+
+/**
+ * Provider neutral {@link Path} comparator.
+ * <p>
+ * Sorting is done via the {@code Path.toUri()} objects.
+ * This is useful when sorting Path objects across FileSystem types.
+ * </p>
+ */
+public class PathComparator implements Comparator<Path>
```

Comment:
Does that make sense to sort by `URI`? If so, I think this class should reflect in its name that it compares by `URI` and possibly it should be moved to the `PathCollators` class.

Reply:
This is a low odds scenario.

But you have a collection consisting of multiple Path roots that are all different java.nio.file.FileSystem types.   Doing the sort by URI keeps the rules the same across the variances of those FileSystems. 

Having this in the PathCollators makes sense too.